accountsservice: fix build for musl

alyssais opened the pull request:
This is now a transitive dependency of flatpak, which was previously buildable for musl.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

alyssais committed:
accountsservice: fix build for musl

This is now a transitive dependency of flatpak, which was previously
buildable for musl.

SuperSandro2000 reviewed (approved):
Maybe we should add a passthru.tests to either package to not regress this in the future?

alyssais committed:
flatpak.tests.musl: init

Would have caught build regression in ae96b319207d ("flatpak: 1.14.10
-> 1.16.0").

getchoo reviewed (commented):
Seems to still fail

```
accountsservice> FAILED: subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/pwd.c.o
accountsservice> gcc -Isubprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p -Isubprojects/mocklibc-1.0/src -I../subprojects/mocklibc-1.0/src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -fPIC -MD -MQ subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/pwd.c.o -MF subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/pwd.c.o.d -o subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/pwd.c.o -c ../subprojects/mocklibc-1.0/src/pwd.c
accountsservice> ../subprojects/mocklibc-1.0/src/pwd.c: In function ‘getpwent’:
accountsservice> ../subprojects/mocklibc-1.0/src/pwd.c:48:10: warning: implicit declaration of function ‘fgetpwent’; did you mean ‘getpwent’? [-Wimplicit-function-declaration]
accountsservice>    48 |   return fgetpwent(global_stream);
accountsservice>       |          ^~~~~~~~~
accountsservice>       |          getpwent
accountsservice> ../subprojects/mocklibc-1.0/src/pwd.c:48:10: error: returning ‘int’ from a function with return type ‘struct passwd *’ makes pointer from integer without a cast [-Wint-conversion]
accountsservice>    48 |   return fgetpwent(global_stream);
accountsservice>       |          ^~~~~~~~~~~~~~~~~~~~~~~~
accountsservice> ../subprojects/mocklibc-1.0/src/pwd.c: In function ‘getpwnam’:
accountsservice> ../subprojects/mocklibc-1.0/src/pwd.c:69:17: error: assignment to ‘struct passwd *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
accountsservice>    69 |   while ((entry = fgetpwent(stream))) {
accountsservice>       |                 ^
accountsservice> ../subprojects/mocklibc-1.0/src/pwd.c: In function ‘getpwuid’:
accountsservice> ../subprojects/mocklibc-1.0/src/pwd.c:90:17: error: assignment to ‘struct passwd *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
accountsservice>    90 |   while ((entry = fgetpwent(stream))) {
accountsservice>       |                 ^
```

<details>
<summary>Full log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/ad8draxgppm6jbwb8hrmzmwrkn210azz-accountsservice-23.13.9.tar.xz
source root is accountsservice-23.13.9
setting SOURCE_DATE_EPOCH to timestamp 1679945259 of file "accountsservice-23.13.9/tests/util.c"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/3m7av8bwpsl032xjrvl9prs30l2sg9lm-fix-paths.patch
patching file src/daemon.c
patching file src/user.c
applying patch /nix/store/afzvsi2i56pycznblzdrh2q57189xw8a-no-create-dirs.patch
patching file meson_post_install.py
applying patch /nix/store/0kp75kx6rpm8yxxzjbax8d2azv2gkv2b-Disable-methods-that-change-files-in-etc.patch
patching file src/daemon.c
patching file src/user.c
applying patch /nix/store/223mwgv177vdxycxklk2sgkscf8l7i9l-drop-prefix-check-extensions.patch
patching file src/extensions.c
applying patch /nix/store/anz90ayg62zzbnw5l4g9kdak889yfpxc-get-dm-type-from-config.patch
patching file src/daemon.c
applying patch /nix/store/4ka0nxxy02yh499l41gqhfj5q3vvpgcp-no-check-wtmp-path.patch
patching file meson.build
patching script interpreter paths in meson_post_install.py
meson_post_install.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/jgj98z09fk5rdapx70a4zi9kc1nmkqdi-python3-3.12.8/bin/python3"
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
mesonConfigurePhase flags: --prefix=/nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9 --libdir=/nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/lib --libexecdir=/nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/libexec --bindir=/nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/bin --sbindir=/nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/sbin --includedir=/nix/store/93ay3dgkm159ic1lv41qmga166g3xbms-accountsservice-23.13.9-dev/include --mandir=/nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/share/man --infodir=/nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/share/info --localedir=/nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/share/locale -Dauto_features=enabled -Dwrap_mode=nodownload --buildtype=plain -Dadmin_group=wheel -Dlocalstatedir=/var -Dsystemdsystemunitdir=/nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/etc/systemd/system
[1mThe Meson build system[0m
Version: 1.6.1
Source dir: [1m/build/accountsservice-23.13.9[0m
Build dir: [1m/build/accountsservice-23.13.9/build[0m
Build type: [1mnative build[0m
Project name: [1maccountsservice[0m
Project version: [1m23.13.9[0m
C compiler for the host machine: [1mgcc[0m (gcc 14.2.1 "gcc (GCC) 14.2.1 20241116")
C linker for the host machine: [1mgcc[0m ld.bfd 2.43.1
Host machine cpu family: [1mx86_64[0m
Host machine cpu: [1mx86_64[0m
Has header "[1mpaths.h[0m" : [1;32mYES[0m 
Has header "[1mshadow.h[0m" : [1;32mYES[0m 
Has header "[1mutmpx.h[0m" : [1;32mYES[0m 
Checking for function "[1mgetusershell[0m" : [1;32mYES[0m 
Checking for function "[1msetutxdb[0m" : [1;31mNO[0m 
Checking for function "[1mfgetpwent[0m" : [1;32mYES[0m 
Header "[1mutmpx.h[0m" has symbol "[1mWTMPX_FILENAME[0m" : [1;31mNO[0m 
Header "[1mpaths.h[0m" has symbol "[1m_PATH_WTMPX[0m" : [1;31mNO[0m 
[1mMessage:[0m whether optimization is enabled: false
Found pkg-config: [1;32mYES[0m [1m(/nix/store/h74c9ad0db9i5bf06071dnmcwdcpfjyf-pkg-config-wrapper-0.29.2/bin/pkg-config)[0m [1;34m0.29.2[0m
Run-time dependency [1mgio-2.0[0m found: [1;32mYES[0m [36m2.82.4[0m
Run-time dependency [1mgio-unix-2.0[0m found: [1;32mYES[0m [36m2.82.4[0m
Run-time dependency [1mglib-2.0[0m found: [1;32mYES[0m [36m2.82.4[0m
Run-time dependency [1mpolkit-gobject-1[0m found: [1;32mYES[0m [36m124[0m
Run-time dependency [1mlibxcrypt[0m found: [1;32mYES[0m [36m4.4.36[0m
Run-time dependency [1mdbus-1[0m found: [1;32mYES[0m [36m1.14.10[0m
Run-time dependency [1mlibsystemd[0m found: [1;32mYES[0m [36m257[0m
Configuring [1morg.freedesktop.Accounts.service[0m using configuration
Program [1mmsgfmt[0m found: [1;32mYES[0m (/nix/store/n80mwf88k123a9m4d5f3w6cwzayynpkl-gettext-0.22.5/bin/msgfmt)
Configuring [1maccounts-daemon.service[0m using configuration
Dependency [1mgio-2.0[0m found: [1;32mYES[0m [36m2.82.4[0m [1;34m(cached)[0m
Program [1m/nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/bin/gdbus-codegen[0m found: [1;32mYES[0m (/nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/bin/gdbus-codegen)
Dependency [1mgio-2.0[0m found: [1;32mYES[0m [36m2.82.4[0m [1;34m(cached)[0m
Program [1m/nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/bin/gdbus-codegen[0m found: [1;32mYES[0m (/nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/bin/gdbus-codegen)
Dependency [1mglib-2.0[0m found: [1;32mYES[0m [36m2.82.4[0m [1;34m(cached)[0m
Program [1m/nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/bin/glib-mkenums[0m found: [1;32mYES[0m (/nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/bin/glib-mkenums)
Dependency [1mglib-2.0[0m found: [1;32mYES[0m [36m2.82.4[0m [1;34m(cached)[0m
Program [1m/nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/bin/glib-mkenums[0m found: [1;32mYES[0m (/nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/bin/glib-mkenums)
Compiler for C supports link arguments -Wl,--version-script,/build/accountsservice-23.13.9/src/libaccountsservice/symbol.map: [1;32mYES[0m 
Run-time dependency [1mgobject-introspection-1.0[0m found: [1;32mYES[0m [36m1.82.0[0m
Dependency [1mgobject-introspection-1.0[0m found: [1;32mYES[0m [36m1.82.0[0m [1;34m(cached)[0m
Program [1m/nix/store/xmacn5c47qzlw973h9x40kxybyxzdkzr-gobject-introspection-1.82.0-dev/bin/g-ir-scanner[0m found: [1;32mYES[0m (/nix/store/xmacn5c47qzlw973h9x40kxybyxzdkzr-gobject-introspection-1.82.0-dev/bin/g-ir-scanner)
Dependency [1mgobject-introspection-1.0[0m found: [1;32mYES[0m [36m1.82.0[0m [1;34m(cached)[0m
Program [1m/nix/store/xmacn5c47qzlw973h9x40kxybyxzdkzr-gobject-introspection-1.82.0-dev/bin/g-ir-compiler[0m found: [1;32mYES[0m (/nix/store/xmacn5c47qzlw973h9x40kxybyxzdkzr-gobject-introspection-1.82.0-dev/bin/g-ir-compiler)
Program [1mvapigen[0m found: [1;32mYES[0m (/nix/store/3da068bpq3844q3qipp4amvh0s5rr9hr-vala-0.56.17/bin/vapigen)
Program [1mmsginit[0m found: [1;32mYES[0m (/nix/store/n80mwf88k123a9m4d5f3w6cwzayynpkl-gettext-0.22.5/bin/msginit)
Program [1mmsgmerge[0m found: [1;32mYES[0m (/nix/store/n80mwf88k123a9m4d5f3w6cwzayynpkl-gettext-0.22.5/bin/msgmerge)
Program [1mxgettext[0m found: [1;32mYES[0m (/nix/store/n80mwf88k123a9m4d5f3w6cwzayynpkl-gettext-0.22.5/bin/xgettext)
Program [1mcheck-format.sh[0m found: [1;32mYES[0m (/build/accountsservice-23.13.9/tests/check-format.sh)
Using [1mmocklibc[0m source from cache.
Using [1mmocklibc[0m patch from cache.

Executing subproject [1mmocklibc[0m 

mocklibc| Project name: [1mmocklibc[0m
mocklibc| Project version: [1m1.0[0m
mocklibc| C compiler for the host machine: [1mgcc[0m (gcc 14.2.1 "gcc (GCC) 14.2.1 20241116")
mocklibc| C linker for the host machine: [1mgcc[0m ld.bfd 2.43.1
mocklibc| Configuring [1mmocklibc[0m using configuration
mocklibc| Program [1minnetgr[0m found: [1;31mNO[0m
mocklibc| Build targets in project: [1m92[0m
mocklibc| Subproject [1mmocklibc[0m finished.

Program [1mpython3[0m found: [1;32mYES[0m (/nix/store/jgj98z09fk5rdapx70a4zi9kc1nmkqdi-python3-3.12.8/bin/python3)
tests/meson.build:29: [1;33mWARNING:[0m Python3 module 'gi' needed by test suite not found
tests/meson.build:75: [1;33mWARNING:[0m Skipping AccuntsService tests as introspection bindings are missing
Program [1msh[0m found: [1;32mYES[0m (/nix/store/5laljj5h14ykan7sl63xs1lca96cn5ny-bash-5.2p37/bin/sh)
Program [1msh[0m found: [1;32mYES[0m (/nix/store/5laljj5h14ykan7sl63xs1lca96cn5ny-bash-5.2p37/bin/sh)
Configuring [1mconfig.h[0m using configuration
[1mMessage:[0m 
accountsservice was configured with the following options:
** DocBook documentation build: false
** Administrator group: wheel
** Extra administrator groups: 
** GDM configuration: /etc/gdm/custom.conf
** LightDM configuration: /etc/lightdm/lightdm.conf
Build targets in project: [1m93[0m
[1mNOTICE:[0m Future-deprecated features used:
 * 0.56.0: {'dependency.get_pkgconfig_variable', 'meson.build_root', 'meson.source_root'}
 * 0.62.0: {'pkgconfig.generate variable for builtin directories'}

accountsservice [36m23.13.9[0m

  [1mSubprojects[0m
    mocklibc            : [1;32mYES[0m

  [1mUser defined options[0m
    admin_group         : wheel
    auto_features       : enabled
    bindir              : /nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/bin
    buildtype           : plain
    includedir          : /nix/store/93ay3dgkm159ic1lv41qmga166g3xbms-accountsservice-23.13.9-dev/include
    infodir             : /nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/share/info
    libdir              : /nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/lib
    libexecdir          : /nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/libexec
    localedir           : /nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/share/locale
    localstatedir       : /var
    mandir              : /nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/share/man
    prefix              : /nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9
    sbindir             : /nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/sbin
    systemdsystemunitdir: /nix/store/dx2rwri07mjz5q6piq5ccgb1ksn2f7a1-accountsservice-23.13.9/etc/systemd/system
    wrap_mode           : nodownload

Found ninja-1.12.1 at /nix/store/j2qsza8wb6ra4nd7qhlixpjkm2d61vkv-ninja-1.12.1/bin/ninja
mesonConfigurePhase: enabled\ parallel\ building
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j12
[1/109] Compiling C object subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/pwd.c.o
FAILED: subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/pwd.c.o 
gcc -Isubprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p -Isubprojects/mocklibc-1.0/src -I../subprojects/mocklibc-1.0/src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -fPIC -MD -MQ subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/pwd.c.o -MF subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/pwd.c.o.d -o subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/pwd.c.o -c ../subprojects/mocklibc-1.0/src/pwd.c
../subprojects/mocklibc-1.0/src/pwd.c: In function ‘getpwent’:
../subprojects/mocklibc-1.0/src/pwd.c:48:10: warning: implicit declaration of function ‘fgetpwent’; did you mean ‘getpwent’? [-Wimplicit-function-declaration]
   48 |   return fgetpwent(global_stream);
      |          ^~~~~~~~~
      |          getpwent
../subprojects/mocklibc-1.0/src/pwd.c:48:10: error: returning ‘int’ from a function with return type ‘struct passwd *’ makes pointer from integer without a cast [-Wint-conversion]
   48 |   return fgetpwent(global_stream);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
../subprojects/mocklibc-1.0/src/pwd.c: In function ‘getpwnam’:
../subprojects/mocklibc-1.0/src/pwd.c:69:17: error: assignment to ‘struct passwd *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
   69 |   while ((entry = fgetpwent(stream))) {
      |                 ^
../subprojects/mocklibc-1.0/src/pwd.c: In function ‘getpwuid’:
../subprojects/mocklibc-1.0/src/pwd.c:90:17: error: assignment to ‘struct passwd *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
   90 |   while ((entry = fgetpwent(stream))) {
      |                 ^
[2/109] Compiling C object subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/grp.c.o
FAILED: subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/grp.c.o 
gcc -Isubprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p -Isubprojects/mocklibc-1.0/src -I../subprojects/mocklibc-1.0/src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -fPIC -MD -MQ subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/grp.c.o -MF subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/grp.c.o.d -o subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/grp.c.o -c ../subprojects/mocklibc-1.0/src/grp.c
../subprojects/mocklibc-1.0/src/grp.c: In function ‘getgrent’:
../subprojects/mocklibc-1.0/src/grp.c:48:10: warning: implicit declaration of function ‘fgetgrent’; did you mean ‘getgrent’? [-Wimplicit-function-declaration]
   48 |   return fgetgrent(global_stream);
      |          ^~~~~~~~~
      |          getgrent
../subprojects/mocklibc-1.0/src/grp.c:48:10: error: returning ‘int’ from a function with return type ‘struct group *’ makes pointer from integer without a cast [-Wint-conversion]
   48 |   return fgetgrent(global_stream);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
../subprojects/mocklibc-1.0/src/grp.c: In function ‘getgrnam’:
../subprojects/mocklibc-1.0/src/grp.c:69:17: error: assignment to ‘struct group *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
   69 |   while ((entry = fgetgrent(stream))) {
      |                 ^
../subprojects/mocklibc-1.0/src/grp.c: In function ‘getgrgid’:
../subprojects/mocklibc-1.0/src/grp.c:90:17: error: assignment to ‘struct group *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
   90 |   while ((entry = fgetgrent(stream))) {
      |                 ^
../subprojects/mocklibc-1.0/src/grp.c: In function ‘getgrouplist’:
../subprojects/mocklibc-1.0/src/grp.c:119:17: error: assignment to ‘struct group *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  119 |   while ((entry = fgetgrent(stream))) {
      |                 ^
[3/109] Compiling C object subprojects/mocklibc-1.0/src/mocklibc-debug-netgroup.p/netgroup-debug.c.o
../subprojects/mocklibc-1.0/src/netgroup-debug.c: In function ‘netgroup_debug_print_entry’:
../subprojects/mocklibc-1.0/src/netgroup-debug.c:25:3: warning: implicit declaration of function ‘print_indent’ [-Wimplicit-function-declaration]
   25 |   print_indent(stream, indent);
      |   ^~~~~~~~~~~~
[4/109] Compiling C object subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/netdb.c.o
[5/109] Merging translations for data/org.freedesktop.accounts.policy
[6/109] Generating gdbus source src/accounts-generated.c
[7/109] Generating gdbus header src/accounts-generated.h
[8/109] Generating gdbus source src/accounts-user-generated.c
[9/109] Generating gdbus header src/accounts-user-generated.h
[10/109] Generating GObject enum file src/libaccountsservice/act-user-enum-types.h (wrapped by meson to capture output)
[11/109] Compiling C object subprojects/mocklibc-1.0/src/libmocklibc.so.0.0.0.p/netgroup.c.o
[12/109] Compiling C object tests/util.p/util.c.o
In file included from ../tests/util.c:3:
../src/util.c: In function ‘compat_check_exit_status’:
../src/util.c:164:9: warning: ‘g_spawn_check_exit_status’ is deprecated: Use 'g_spawn_check_wait_status' instead [-Wdeprecated-declarations]
  164 |         return g_spawn_check_exit_status (estatus, error);
      |         ^~~~~~
In file included from /nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/include/glib-2.0/glib.h:84,
                 from /nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/include/glib-2.0/gobject/gbinding.h:30,
                 from /nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/include/glib-2.0/glib-object.h:24,
                 from /nix/store/6mbf98z1vs3fx28vxcy5s2xw31cxaxd9-polkit-124-dev/include/polkit-1/polkit/polkittypes.h:25,
                 from /nix/store/6mbf98z1vs3fx28vxcy5s2xw31cxaxd9-polkit-124-dev/include/polkit-1/polkit/polkit.h:27,
                 from ../src/util.c:33:
/nix/store/paq90ba9iz22db5d6a8jkybf88gbk9hy-glib-2.82.4-dev/include/glib-2.0/glib/gspawn.h:315:10: note: declared here
  315 | gboolean g_spawn_check_exit_status (gint      wait_status,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

</details>
